demos/common/utilities/jws_using_mbedtls.c

```diff
@@ -0,0 +1,634 @@
+ * @param ulNLength The length of \p pucN.
+ * @param pucE The exponent used for the key.
+ * @param ulELength The length of \p pucE.
+ * @param pucBuffer The buffer used as scratch space to make the calculations. It should be at least
+ * `jwsRSA3072_SIZE` + `jwsSHA256_SIZE` in size.
+ * @param ulBufferLength The length of \p pucBuffer.
+ * @return uint32_t The result of the operation.
+ * @retval 0 if successful.
+ * @retval Non-0 if not successful.
+ */
+static uint32_t prvJWS_RS256Verify( unsigned char * pucInput,
+                                    uint32_t ulInputLength,
+                                    unsigned char * pucSignature,
+                                    uint32_t ulSignatureLength,
+                                    unsigned char * pucN,
+                                    uint32_t ulNLength,
+                                    unsigned char * pucE,
+                                    uint32_t ulELength,
+                                    unsigned char * pucBuffer,
+                                    uint32_t ulBufferLength )
+{
+    AzureIoTResult_t xResult;
+    int32_t lMbedTLSResult;
+
+    if( ulBufferLength < jwsRSA3072_SIZE + jwsSHA256_SIZE )
+    {
+        LogInfo( ( "Buffer Not Large Enough\n" ) );
+        return 1;
+    }
+
+    unsigned char * pucShaBuffer = pucBuffer + jwsRSA3072_SIZE;
+    size_t ulDecryptedLength;
+
+    /* The signature is encrypted using the input key. We need to decrypt the */
+    /* signature which gives us the SHA256 inside a PKCS7 structure. We then compare */
+    /* that to the SHA256 of the input. */
+    mbedtls_rsa_context ctx;
+
+    mbedtls_rsa_init( &ctx, MBEDTLS_RSA_PKCS_V15, 0 );
+
+    LogInfo( ( "--- Initializing Decryption ---\n" ) );
+
+    lMbedTLSResult = mbedtls_rsa_import_raw( &ctx,
+                                             pucN, ulNLength,
+                                             NULL, 0,
+                                             NULL, 0,
+                                             NULL, 0,
+                                             pucE, ulELength );
+
+    if( lMbedTLSResult != 0 )
+    {
+        LogError( ( "mbedtls_rsa_import_raw res: %i\n", lMbedTLSResult ) );
+        mbedtls_rsa_free( &ctx );
+        return lMbedTLSResult;
+    }
+
+    lMbedTLSResult = mbedtls_rsa_complete( &ctx );
+
+    if( lMbedTLSResult != 0 )
+    {
+        LogError( ( "mbedtls_rsa_complete res: %i\n", lMbedTLSResult ) );
+        mbedtls_rsa_free( &ctx );
+        return lMbedTLSResult;
+    }
+
+    lMbedTLSResult = mbedtls_rsa_check_pubkey( &ctx );
+
+    if( lMbedTLSResult != 0 )
+    {
+        LogError( ( "mbedtls_rsa_check_pubkey res: %i\n", lMbedTLSResult ) );
+        mbedtls_rsa_free( &ctx );
+        return lMbedTLSResult;
+    }
+
+    LogInfo( ( "--- Decrypting ---\n" ) );
+
+    /* RSA */
+    lMbedTLSResult = mbedtls_rsa_pkcs1_decrypt( &ctx, NULL, NULL, MBEDTLS_RSA_PUBLIC, &ulDecryptedLength, pucSignature, pucBuffer, jwsRSA3072_SIZE );
```

Comment:
My read of `mbedtls_rsa_pkcs1_decrypt` docs [here](https://github.com/Mbed-TLS/mbedtls/blob/ca3c6a5698ef728bcede4a3db2bee964ee489a45/include/mbedtls/rsa.h#L696) is that the 2nd param, `f_rng` is mandatory.  Are MBED docs off or if not, what's consequence of this bein NULL?

 (I'm all for it being NULL FWIW to keep our life easier but I don't understand impact.)